index.js

```diff
@@ -1,10 +1,12 @@
 const execSync = require('child_process').execSync
 
 function getProcesses (command) {
-  const lsofOutput = execSync(`lsof -i TCP -P -n | grep '${command}\\s.*:[0-9]* (LISTEN)' | cat`, {encoding: 'utf-8'})
-    .toString()
-    .split('\n')
   const processesObject = {}
+  const regex = /[^A-Za-z0-9 .\-_]+/g
```

Comment:
The regexp doesn't seem to work with a potentially legit input.
The code line `"service".match(/[^A-Za-z0-9 .\-_]+/g)` return `null`.

Reply:
Hi @toufik-airane,
```javascript
if(command.match(regex)){
    return processesObject
  }
```
The if condition is supposed to catch only illegal characters. Completely legit inputs would go with else condition. So it's works Perfectly.